api/src/main/java/org/pmiops/workbench/captcha/CaptchaVerificationServiceImpl.java

```diff
@@ -60,11 +91,20 @@ public boolean verifyCaptcha(String responseToken) throws ApiException {
     }
     String captchaHostname = response.getHostname();
     String uiUrl = configProvider.get().admin.loginUrl;
+    boolean usingTestCaptcha = configProvider.get().captcha.useTestCaptcha;
+    boolean captchaHostNameMatchUI = false;
 
-    // check if the UI URL has the host as send by Captcha Response
-    boolean captchaHostNameMatchUI =
-        String.format(urlPattern, captchaHostname).equals(uiUrl)
-            || String.format(localHostUrlPattern, captchaHostname).equals(uiUrl);
+    // Production/Stable should  not use google test Key and the domainName should match with the
+    // one return from Captcha Response
+    if (!usingTestCaptcha) {
+      // check if the UI URL has the host as send by Captcha Response
+      captchaHostNameMatchUI = String.format(urlPattern, captchaHostname).equals(uiUrl);
+    } else {
+      // If URL is using google test key hence the host return will always be testkey.google.com
+      // hence check if the UI url in config matches one of AllOfUs urls(local/test/staging/perf)
+      captchaHostNameMatchUI =
+          captchaHostname.equals(googleTestHost) && testCaptchaURLs.contains(uiUrl);
```

Comment:
i have added the logic again for testing the urls that are using test google captha. Since the host from the captcha service will always be testkey.google.com, check if the ui url will be one of the environment using test captcha test/local/staging and perf. Let me know if this makes sense/ is there any other check that i can add.

Reply:
Sorry, I still don't understand why you need to hardcode a list of URLs here. Can you not use the combination of `config.server.uiBaseUrl` with `usingTestCaptcha` to make the determination?